frontend/lib/src/util/getBrowserInfo.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022-2024)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import UAParser from "ua-parser-js"
+
+function getBrowserInfo(): {
+  browserName: string
+  browserVersion: string
+  deviceType: string
+  os: string
+} {
+  const parser = new UAParser()
+  const result = parser.getResult()
+
+  return {
+    browserName: result.browser.name || "Unknown",
+    browserVersion: result.browser.version || "Unknown",
+    deviceType: result.device.type || "desktop",
```

Comment:
Why do we use `desktop` here as fallback and not `Unknown`? Is it only filled if used by mobile?

Reply:
from my tinkering around in the unit tests, that was the behavior i observed. i could see the argument that we should just leave it as 'Unknown' or 'mobile' but felt 'desktop' made things a bit more clear. no strong opinions either way from me.

Reply:
Ok 👍  the docs also mention that `desktop` cannot be detected: 

<img width="817" alt="image" src="https://github.com/user-attachments/assets/0e45c670-034c-4b6a-b2c7-f0311dc822c0">

I think it's fine to fall back to `desktop`. But maybe add a quick comment there similar to the comment in ua-parsers-js about desktop.

Reply:
sounds good!

Reply:
wonder if we'll see any 'wearable' or 'console' streamlit app viewership 🤔